core/arch/instr.h

```diff
@@ -326,7 +326,7 @@ typedef enum _dr_encoding_hint_type_t {
 } dr_encoding_hint_type_t;
 /* DR_API EXPORT END */
 
-#define DR_TUPLE_TYPE_BITS 16
+#define DR_TUPLE_TYPE_BITS 13
```

Comment:
These aren't used like bitmasks: they are mutually exclusive, so they can fit in 4 bits as regular integers.  That makes more sense with how they're used: you're using case statements, which seems unusual if they're power-of-2 values.  That would also match the pred encoding in the upper bits of instr_t.prefixes.

Reply:
Fixed.